src/MerQure.Tools/Buses/RetryConsumer.cs

```diff
@@ -0,0 +1,98 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using MerQure;
+using MerQure.Tools.Messages;
+using MerQure.Tools.Configurations;
+using Newtonsoft.Json;
+using MerQure.Messages;
+using MerQure.Tools.Exceptions;
+
+namespace MerQure.Tools.Buses
+{
+    internal class RetryConsumer<T> : Consumer<T> where T : IDelivered
+    {
+        private readonly Publisher<T> _producer;
+
+        private RetryStrategyConfiguration _retryConfiguration;
+
+        public RetryConsumer(IMessagingService messagingService, RetryStrategyConfiguration retryConfiguration, Publisher<T> producer) : base(messagingService)
+        {
+            _retryConfiguration = retryConfiguration;
+            _producer = producer;
+        }
+
+        internal void SendToErrorExchange(Channel channel, T deliveredMessage)
+        {
+            if (String.IsNullOrEmpty(deliveredMessage.DeliveryTag))
+                throw new ArgumentNullException(nameof(deliveredMessage.DeliveryTag));
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') The parameter name 'DeliveryTag' is not declared in the argument list. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS3928)